core/src/main/java/com/testaem/aem/core/utills/impl/HTMLTruncator.java

```diff
@@ -0,0 +1,72 @@
+package com.testaem.aem.core.utills.impl;
+
+import com.testaem.aem.core.utills.Truncator;
+
+import java.util.*;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class HTMLTruncator implements Truncator {
+
+    private static final Pattern WORDS_PATTERN = Pattern.compile("&.*?;|<.*?>|(\\w[\\w-]*)");
+    private static final Pattern TAGS_PATTERN = Pattern.compile("<(/)?([^ ]+?)(?: | .*?)?(/)?>");
+    private static final List<String> HTML_FOR_SINGLETS = Arrays.asList("br", "col", "link", "base", "img", "param", "area", "hr", "input");
+
+    private static String truncateHtmlWords(String html, int length, String end) {
```

Comment:
Great! Cool!)
But consider splitting into several methods to reduce the cognitive complexity of the code 

Also, the code is too nested. They usually welcome no more than two nested levels like
```
while () {
   if () {
   }
}
```
or 
```
try (
   while() {
   }
} catch {}

... and `if`-s without `{}` are still awful) Especially when mixed up with if-s having `{}`